snapcraft_legacy/internal/remote_build/_launchpad.py

```diff
@@ -193,12 +193,32 @@ def _issue_build_request(self, snap: Entry) -> Entry:
         )
 
     def _lp_load_url(self, url: str) -> Entry:
-        """Load Launchpad url with a retry in case the connection is lost."""
-        try:
-            return self._lp.load(url)
-        except ConnectionResetError:
-            self._lp = self.login()
-            return self._lp.load(url)
+        """Load Launchpad url with an exponential backoff retry in case the connection is lost."""
+        max_retries = 5
+        base_delay = 1
+        retry_count = max_retries
+
+        while retry_count > 0:
+            try:
+                return self._lp.load(url)
+            except ConnectionError as conn_error:
+                logger.debug(
+                    "Connection error when loading the URL "
+                    "{!r} (retries left {})".format(url, retry_count)
+                )
+
+                if retry_count == 1:
+                    raise conn_error
+
+                self._lp = self.login()
+
+                delay = base_delay * (2 ** (max_retries - retry_count))
+                time.sleep(delay)
```

Comment:
What do you think of adding a logger.debug message like `Failed to load URL.  Waiting <x> seconds before retrying.`?